src/stochprog.jl

```diff
@@ -0,0 +1,112 @@
+using LightGraphs
+
+export AbstractStochasticProgram, stochasticprogram
+export isleaf, probability, cutgenerator, numberofpaths
+export setprobability!, add_scenario_state!, add_scenario_transition!
+
+"""
+    AbstractStochasticProgram
+
+Stochastic program instance
+"""
+abstract type AbstractStochasticProgram <: AbstractGraph{Int} end
+
+"""
+    stochasticprogram(args...)
+
+Creates a stochastic program from the arguments
+"""
+function stochasticprogram end
+
+"""
+    getmaster(sp::AbstractStochasticProgram)
+
+Returns the master node of stochastic problem `sp`.
+"""
+function getmaster end
+
+"""
+    probability(sp::AbstractStochasticProgram, edge)
+
+Returns the probability to take the edge `edge` in the stochastic problem `sp`.
+"""
+function probability end
+
+"""
+    setchildx!(sp::AbstractStochasticProgram, node, child, sol)
+
+Sets the parent solution of `child` as `sol`, the solution obtained at `node`.
+"""
+function setchildx! end
```

Comment:
I'm not sure what this means? Should it be `setchildstate!(sp, parent, child)`?

Reply:
Yes the name is awful and the argument list is even worse :D 
For some obscure reason I need to also give the parent as parameter (because of the `childT` I used for the [EntropicCone](https://github.com/blegat/EntropicCone.jl) to account for the permutation tranformations).
I prefer `setchildstate!(sp, child, sol)` because in `setchildstate!(sp, parent, child)`, we need to get the `outgoing state` of `parent` but in the mean time it might have been solved by another thread and the solution might not be the one we want.

Reply:
Then it should just be `setincomingstate!(sp, child, state::Vector{T})`?

Reply:
In case the parent is unbounded, the solution also contain an unbounded ray (which is composed of a feasible point and an unbounded vector). In this case, passing just a vector is not enough. Passing an `AbstractSolution` provides enough flexibility to handle this case.